eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -114,7 +114,7 @@ stages:
           parameters:
             # Flag to include the job template with a container field
             ${{ each platform in parameters.AdditionalPlatforms }}:
-              ${{ if contains(coalesce(platform.value.SupportedClouds, parameters.Clouds), cloud.key) }}:
+              ${{ if contains(coalesce(platform.value.SupportedClouds, cloud.key), cloud.key) }}:
```

Comment:
I'm not sure I understand this change. Wouldn't this always return true in the case the SupportedClouds isn't set? Also are there any cases were we don't expect SupportedClouds to be set?

Reply:
The problem is the addition of the tests-weekly change I made. So with the current code, this scenario exists:

Build.DefinitionName contains tests-weekly
platforms.value.SupportedClouds == null
parameters.Clouds == "Public"

So we end up evaluating something like the below as false when it should be true:

```
contains(coalesce("", "Public"), "UsGov") == false
```

Another way I could write this change would be below. We don't actually need to check `parameters.Clouds` since we do at a higher level conditional in this same stage definition.

```
${{ if or(contains(platform.value.SupportedClouds, cloud.key), contains(variables['Build.DefinitionName'], 'tests-weekly')) }}
```

Reply:
Ok I believe I understand this but it certainly a little strange.

Reply:
Ok I think I made the expression much clearer now:

```
${{ if or(not(platform.value.SupportedClouds), contains(platform.value.SupportedClouds, cloud.key)) }}
```

Reply:
Yes that seems less magical :)